projects/python-client/src/datapane/client/api/user.py

```diff
@@ -113,6 +116,76 @@ def _run_script(script: str):
     runpy.run_path(str(script_path), run_name="__datapane__")
 
 
+def _run_template(template_path: str):
+    """Run the template script"""
+    # Store the cwd before we change it
+    old_cwd = os.getcwd()
+    # Change cwd to support relative paths in template script
+    os.chdir(template_path)
+
+    try:
+        runpy.run_path(
+            "template.py",
+            run_name="__datapane__",
+        )
+    # Notify the user of missing packages that are required by the template
+    except ModuleNotFoundError as e:
+        raise DPError(f"Please install the following packages to run this template\n{e.name}") from e
+
+    os.chdir(old_cwd)
+
+
+def _download_template(url: str):
+    """Download the template from a repository url, and delete the .git directory
+
+    Returns:
+        The path of the cloned template repository
+    """
+
+    url = _check_repo_url(url)
+
+    # Check if target directory already exists
+    target = url.split("/")[-1]
+    target_dir_exists = os.path.exists(target)
```

Comment:
generally better to use `Path` for these type of operations rather than `os.path` going forwards

Reply:
`os.path` has been used here to mirror `dulwich.porcelain.clone`'s target directory check [1]. If possible, I'd like to keep this particular instance of `os.path` usage identical to the check that comes after. I've updated the comment with this info.

[1] https://github.com/jelmer/dulwich/blob/78e6ae0960d79060d4ff19f0aa5dc4b32296116d/dulwich/porcelain.py#L439-L441

Reply:
Doesn't this end up making a directory with `.git` appended on the end when using the full repository names?

Reply:
@antking it does - tested with

poetry run datapane template https://github.com/datapane/dp-template-classifier-dashboard.git
poetry run datapane template https://github.com/datapane/dp-template-classifier-dashboard
poetry run datapane template dp-template-classifier-dashboard